tests/components/lcn/test_device_trigger.py

```diff
@@ -0,0 +1,409 @@
+"""Tests for LCN device triggers."""
+from unittest.mock import patch
+
+from pypck.inputs import ModSendKeysHost, ModStatusAccessControl
+from pypck.lcn_addr import LcnAddr
+from pypck.lcn_defs import AccessControlPeriphery, KeyAction, SendKeyCommand
+import voluptuous_serialize
+
+from homeassistant.components import automation
+from homeassistant.components.lcn import device_trigger
+from homeassistant.components.lcn.const import DOMAIN, KEY_ACTIONS, SENDKEYS
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_PLATFORM, CONF_TYPE
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.setup import async_setup_component
+
+from .conftest import MockPchkConnectionManager, get_device, init_integration
+
+from tests.common import assert_lists_same, async_get_device_automations
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_get_triggers_module_device(hass, entry):
+    """Test we get the expected triggers from a LCN module device."""
+    await init_integration(hass, entry)
+    device = get_device(hass, entry, (0, 7, False))
+
+    expected_triggers = [
+        {
+            CONF_PLATFORM: "device",
+            CONF_DOMAIN: DOMAIN,
+            CONF_TYPE: "transmitter",
+            CONF_DEVICE_ID: device.id,
+        },
+        {
+            CONF_PLATFORM: "device",
+            CONF_DOMAIN: DOMAIN,
+            CONF_TYPE: "transponder",
+            CONF_DEVICE_ID: device.id,
+        },
+        {
+            CONF_PLATFORM: "device",
+            CONF_DOMAIN: DOMAIN,
+            CONF_TYPE: "fingerprint",
+            CONF_DEVICE_ID: device.id,
+        },
+        {
+            CONF_PLATFORM: "device",
+            CONF_DOMAIN: DOMAIN,
+            CONF_TYPE: "send_keys",
+            CONF_DEVICE_ID: device.id,
+        },
+    ]
+
+    triggers = await async_get_device_automations(hass, "trigger", device.id)
+    assert_lists_same(triggers, expected_triggers)
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_get_triggers_non_module_device(hass, entry):
+    """Test we get the expected triggers from a LCN non-module device."""
+    not_included_types = ("transmitter", "transponder", "fingerprint", "send_keys")
+
+    await init_integration(hass, entry)
+    device_registry = dr.async_get(hass)
+    for device_id in device_registry.devices:
+        device = device_registry.async_get(device_id)
+        if device.model.startswith(("LCN host", "LCN group", "LCN resource")):
+            triggers = await async_get_device_automations(hass, "trigger", device_id)
+            for trigger in triggers:
+                assert trigger[CONF_TYPE] not in not_included_types
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_if_fires_on_transponder_event(hass, calls, entry):
```

Comment:
Why aren't we passing the manager patch as a test parameter?

Reply:
`PchkConnectionManager` is instantiated in `async_setup_entry` within `__init___.py`. For several tests I need exactly this created instance (which itself takes values from `entry.data`). Because of this I am not sure how to access it in my tests... I don't see how I can achieve this using fixtures.

The integration itself stores the connection instance in `hass.data` for later access. I was told not to use `hass.data` in my tests. So I had to  think about another way how to access the instantiated connection. My previous idea was to emulate a   Mock object and store the instance in the `return_value` attribute. Actually that was a hack.

I tried to implement another way in #61719. Maybe that's better?